messaging/src/main/java/edu/tamu/weaver/messaging/config/MessagingConfig.java

```diff
@@ -0,0 +1,44 @@
+package edu.tamu.weaver.messaging.config;
+
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jms.config.DefaultJmsListenerContainerFactory;
+import org.springframework.jms.config.JmsListenerContainerFactory;
+import org.springframework.jms.connection.CachingConnectionFactory;
+import org.springframework.jms.core.JmsTemplate;
+
+@Configuration
+public class MessagingConfig {
+
+    @Value("${spring.activemq.broker-url}")
+    private String brokerUrl;
+    
+    @Bean
+    public CachingConnectionFactory cachingConnectionFactory() {
+        return new CachingConnectionFactory(defaultActiveMQConnectionFactory());
+    }
+    
+    @Bean
+    public JmsTemplate jmsTopicTemplate() {
+        JmsTemplate jmsTemplate = new JmsTemplate(cachingConnectionFactory());
+        jmsTemplate.setPubSubDomain(true);
+        return jmsTemplate;
+    }
+
+    public ActiveMQConnectionFactory defaultActiveMQConnectionFactory() {
```

Comment:
This probably should be private.

Reply:
https://github.com/TAMULib/Weaver-Webservice-Core/commit/904d136088dea9c2fe83658d7ea2401af5fe723c